paint/data/dataset_splits.py

```diff
@@ -0,0 +1,342 @@
+    output_dir : pathlib.Path
+        Output directory for saving the dataset splits.
+    remove_unused_data : bool
+        Whether unused metadata should be removed or not.
+
+    Methods
+    -------
+    get_dataset_splits()
+        Get the dataset splits and save them to a CSV file.
+    """
+
+    def __init__(
+        self,
+        input_file: Union[Path, str],
+        output_dir: Union[Path, str],
+        remove_unused_data: bool = True,
+    ):
+        """
+        Initialize the dataset splitter.
+
+        Parameters
+        ----------
+        input_file : Union[Path, str]
+            Path to the input file containing the metadata required to generate the dataset splits.
+        output_dir : Union[Path, str]
+            Path to the output directory where the dataset splits will be saved.
+        remove_unused_data : bool
+            Whether extra metadata should be removed from the resulting dataset splits (Default: `True`).
+        """
+        input_file = Path(input_file)
+        if not input_file.exists():
+            raise FileNotFoundError(
+                f"Input file containing metadata does not exist at {input_file}"
+            )
+        self.metadata = pd.read_csv(input_file, index_col=mappings.ID_INDEX)
+        self.metadata[mappings.DATETIME] = pd.to_datetime(
+            self.metadata[mappings.DATETIME]
+        ).dt.tz_localize(None)
+        self.metadata = self.metadata.drop(
+            columns=[
+                "lower_left_latitude",
+                "lower_left_longitude",
+                "lower_left_Elevation",
+                "upper_left_latitude",
+                "upper_left_longitude",
+                "upper_left_Elevation",
+                "upper_right_latitude",
+                "upper_right_longitude",
+                "upper_right_Elevation",
+                "lower_right_latitude",
+                "lower_right_longitude",
+                "lower_right_Elevation",
+            ]
+        )
+        self.metadata[mappings.SPLIT_KEY] = ""
+        self.output_dir = Path(output_dir)
+        self.output_dir.mkdir(parents=True, exist_ok=True)
+        self.remove_unused_data = remove_unused_data
+
+    @staticmethod
+    def _get_azimuth_splits(
+        heliostat_data: pd.DataFrame, training_size: int, validation_size: int
+    ) -> pd.DataFrame:
+        """
+        Get the splits for the azimuth-based splitting method.
+
+        For a single heliostat, the ``training_size`` indices with the smallest azimuth values are selected for the
+        training split, while the ``validation_size`` indices with the largest values are selected for the validation
+        split. The remaining indices are assigned to the test split.
+
+        This ensures that indices with very different azimuth values are considered in the train and validation samples,
+        i.e. the train and validation splits should contain very different samples. This difference leads to a high
```

Comment:
```suggestion
        i.e., the train and validation splits should contain very different samples. This difference leads to a high
```
For American English consistency :).